Add zkEVM vs zkVM blog post (#160)
risc0/website: blog/2023-05-XX-zkevm-vs-zkvm/index.md

---
slug: zkVM
title: 'zkEVM vs zkVM: a Single Letter Can Make a Big Difference!'

> authors: [erik] tags: [zkEVM, zkVM]

Reviewer: `tags` needs its own line
```suggestion
authors: [erik]
tags: [zkEVM, zkVM]
```

tags: [zkEVM, zkVM]
---

> As the current smart contract landscape increases in complexity, developer are

Reviewer: ```suggestion
As the current smart contract landscape increases in complexity, developers are
```

> As the current smart contract landscape increases in complexity, developers are
> finding that the on-chain code they maintain will only increase in cost as they
> develop it. In other words, if your DeFi logic is becoming

Reviewer: ```suggestion
As the current smart contract landscape increases in complexity, with dapps
becoming more advanced and blockspace more scarce, the costs of running on-chain
code are only increasing. In other words, if your DeFi logic is becoming
```

that programmers for the zkVM can use relevant libraries that others have
developed within the language’s ecosystem. One example is that we can run
programs that play games like
[Wordle](https://github.com/risc0/risc0/tree/main/examples/wordle) or [Where’s
Waldo](https://github.com/risc0/risc0/tree/main/examples/waldo). However, we
aren’t limited to simple games: you can run many other programs that compile to
RISC-V.

## What’s so useful about running RISC-V programs?

If you started reading about this because you wanted to learn more about
frameworks that can help you write better DeFi apps, you might be wondering why
we are working on such a generic computation framework. By allowing users to
write programs using general-purpose languages, we open up the possibilities to
writing code outside of Solidity libraries. This means that the application
logic does not need to be limited to what can be expressed using Solidity and
allow you to write chain-agnostic code. General-purpose languages like Rust
allows developers to write different kinds of programs more easily than
languages that are designed for a specific purpose. For example, you can write

> a simple arithmetic calculator using rust that takes a mathematical expression

Reviewer: ```suggestion
a simple arithmetic calculator using Rust that takes a mathematical expression
```

EVM that is written in Rust. By using this crate, we can run an EVM bytecode
interpreter on the zkVM. This means that you can run solidity contracts on the
EVM bytecode interpreter that runs on the zkVM! By doing so, the zkVM generates
a receipt that represents execution of the EVM running a smart contract. This
has been explored in [this blog post](https://odra.dev/blog/evm-at-risc0/) by
Odra and has been used by the [zkPoEx team to improve bug
bounties](https://risczero.com/blog/zkpoex).

## How to scale blockchains

As the current smart contract landscape increases in complexity, many people
find that the on-chain code they maintain will only increase in cost as they
develop it. In other words, if your DeFi logic is getting to become
computationally expensive or your on-chain game logic needs the extra compute
power, RISC Zero’s zkVM and the upcoming Bonsai network allow you to write
scalable chain-agnostic code using general purpose programming languages! I’ve
listed a few examples in this blog post, and we are at the very beginning of
truly scalable computation network. If you want to see what this is all about,
[get started today](https://www.risczero.com/docs), [join our

> community][https://discord.gg/risczero], sign up for bonsai and help build a

Reviewer: Use `()` not `[]` for this (the syntax here is `()` means "external link" whereas `[]` means "reference link": https://www.markdownguide.org/basic-syntax/#reference-style-links).

```suggestion
community](https://discord.gg/risczero), sign up for bonsai and help build a
```

---
slug: zkVM
title: 'zkEVM vs zkVM: a Single Letter Can Make a Big Difference!'
authors: [erik]
tags: [zkEVM, zkVM]
---

> Earlier this month, I had opportunity to attend several conferences. During

Reviewer 2: "earlier this month" is out of date

Author: You're right! It's May already! Time flies when you zk

Reviewer 2: Perhaps something like: 

`As the current smart contract landscape increases in complexity, developer are finding that the on-chain code they maintain will only increase in cost as they develop it. In other words, if your DeFi logic is getting to become computationally expensive or your on-chain game logic needs the extra compute power, you should be looking at using a zkVM or zkEVM to move the complex application logic off-chain.

In this blogpost, I want to explain the difference a single letter can make in this context. I’ll start with an overview of EVM, followed by zkEVM, and move on to zkVM. While zkEVMs allow you to transport Solidity applications off-chain, RISC Zero’s zkVM and the upcoming Bonsai network allow you to write scalable, chain-agnostic code using Rust.`

Author: Thanks! I was going for a blend of personal anecdote and problems in the space but I think your suggestion makes everything more straight-forward

tags: [zkEVM, zkVM]
---

As the current smart contract landscape increases in complexity, developer are
finding that the on-chain code they maintain will only increase in cost as they

> develop it. In other words, if your DeFi logic is getting to become

Reviewer: ```suggestion
develop it. In other words, if your DeFi logic is becoming
```
